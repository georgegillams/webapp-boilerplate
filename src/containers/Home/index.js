import React, { memo } from 'react';
import PropTypes from 'prop-types';

import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { useInjectReducer } from 'utils/inject-reducer';
import { useInjectSaga } from 'utils/inject-saga';

import Features from 'components/Features';
import Showcases from 'components/Showcases';

import saga from './saga';
import reducer from './reducer';
import { getShowcases } from './actions';
import { selectShowcases } from './selectors';

export function Home({ getShowcases, showcasesData }) {
  useInjectSaga({ key: 'showcases', saga });
  useInjectReducer({ key: 'showcases', reducer });

  return (
    <div>
      <Features />
      <Showcases onGetShowcases={getShowcases} data={showcasesData} />
    </div>
  );
}

const mapStateToProps = createStructuredSelector({
  showcasesData: selectShowcases(),
});

export function mapDispatchToProps(dispatch) {
  return { getShowcases: () => dispatch(getShowcases()) };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);

Home.propTypes = {
  showcasesData: PropTypes.object,
  getShowcases: PropTypes.func,
};

export default compose(withConnect, memo)(Home);
