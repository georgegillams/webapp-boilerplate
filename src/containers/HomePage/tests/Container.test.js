import React from 'react';
import { render } from '@testing-library/react';

import Home from '../Container';

describe('<Home />', () => {
  it('should render correctly', () => {
    global.Date.getMonth = jest.fn(() => 2);
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });
});
