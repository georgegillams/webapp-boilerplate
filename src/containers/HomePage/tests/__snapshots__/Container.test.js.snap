// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<div>
  <div
    class="home-page__container"
  >
    <div
      class="typography__main"
    >
      <h1
        class="typography__text typography__text--section page-title__section"
        id="home"
      >
        Home
      </h1>
      <div
        class="typography__main"
      >
        <span
          class="paragraph"
        >
          This is just a boilerplate page, but you can put any content you want in here.
          <br />
          This page is pretty boring, but the 
          <a
            class="text-link text-link--text-link text-link--no-padding text-link--inline"
            href="/sitemap"
          >
            site map
          </a>
           might be more inspiring.
        </span>
      </div>
    </div>
  </div>
</div>
`;
